lib/voucherify/service/redemptions.rb

```diff
@@ -10,7 +10,13 @@ def initialize(client)
       end
 
       def redeem(code, params = {})
-        @client.post("/vouchers/#{URI.encode(code)}/redemption", params.to_json)
+        endpoint = nil
+        if code.is_a? Hash
+          endpoint = "/promotions/tiers/#{URI.encode(code[:id] || code['id'])}/redemption"
+        elsif code.is_a? String
```

Comment:
this `elsif` is bugging me, I would either remove the check or check whether the endpoint is null. I'd prefer the first one as it will not lead to not setting the endpoint